src/lib.rs

```diff
@@ -310,84 +310,75 @@ where
     /// the in-memory changes will be discarded.
     pub fn revert_to(
         &mut self,
-        _requested_id: ChangeID,
+        requested_id: ChangeID,
+        current_id: ChangeID,
     ) -> Result<(), BonsaiStorageError<DB::DatabaseError>> {
-        // self.tries.reset_to_last_commit()?;
-
-        // let kv = self.tries.db_mut();
-
-        // // Clear current changes
-        // kv.changes_store.current_changes.0.clear();
-
-        // // If requested equals last recorded, do nothing
-        // if Some(&requested_id) == kv.changes_store.id_queue.back() {
-        //     return Ok(());
-        // }
-
-        // // Make sure we are not trying to revert with an invalid id
-        // let Some(id_position) = kv
-        //     .changes_store
-        //     .id_queue
-        //     .iter()
-        //     .position(|id| *id == requested_id)
-        // else {
-        //     return Err(BonsaiStorageError::GoTo(format!(
-        //         "Requested id {:?} was removed or has not been recorded",
-        //         requested_id
-        //     )));
-        // };
-
-        // // Accumulate changes from requested to last recorded
-        // let mut full = Vec::new();
-        // for id in kv
-        //     .changes_store
-        //     .id_queue
-        //     .iter()
-        //     .skip(id_position)
-        //     .rev()
-        //     .take_while(|id| *id != &requested_id)
-        // {
-        //     full.extend(
-        //         ChangeBatch::deserialize(
-        //             id,
-        //             kv.db.get_by_prefix(&DatabaseKey::TrieLog(&id.to_bytes()))?,
-        //         )
-        //         .0,
-        //     );
-        // }
+        self.tries.reset_to_last_commit()?;
+
+        let kv = self.tries.db_mut();
+
+        // Clear current changes
+        kv.changes_store.current_changes.0.clear();
+
+        let revert_to_id = requested_id.as_u64();
+        let latest_id = current_id.as_u64();
+        let next_id = latest_id.saturating_add(1);
+
+        // ensure that the id is the latest by checking for an id one higher
+        // note that we don't use contains() because we have a prefix, not the full key
+        let next_id = ChangeID::from_u64(next_id);
+        if let Ok(matches) = kv
```

Comment:
i think it's okay to remove this check and just put that invariant in the doc comment of the function
In madara we can make sure it's always true without this weird thing